x/mongo/driver/topology/CMAP_spec_test.go

```diff
@@ -162,7 +162,7 @@ func runCMAPTest(t *testing.T, testFileName string) {
 			for len(testInfo.backgroundThreadErrors) > 0 {
 				bgErr := <-testInfo.backgroundThreadErrors
 				errs = append(errs, bgErr)
-				if bgErr != nil && strings.ToLower(test.Error.Message) == bgErr.Error() {
+				if bgErr != nil && strings.Contains(bgErr.Error(), strings.ToLower(test.Error.Message)) {
```

Comment:
If it is straightforward, I suggest adding a test to check that `mongo.IsTimeout()` returns true on the WaitQueueTimeoutError. That would protect against a future regression later if pooling code changes.

I think this could be done with a unit test in pool_test.go:
- create a pool of maxPoolSize=1
- `get` a connection
- `get` with a small timeout
- assert that the returned error satisfies `mongo.IsTimeout`

Reply:
It does seem smart to have a unit test, so I added one to `pool_test.go`. It does not use `mongo.IsTimeout(...)` because that results in a circular import. Instead, I check that unwrapping the error produces a `context.DeadlineExceeded`, and that it is in fact an instance of `WaitQueueTimeoutError`.

Reply:
Awesome! Ah I had not considered the circular import.